core/src/main/java/org/pentaho/platform/dataaccess/datasource/ui/selectdialog/DatasourceSelectionDialogController.java

```diff
@@ -469,6 +473,12 @@ public void onDialogError( String value ) {
 
   @Bindable
   public void removeDatasourceConfirm() {
+    if ( messageBundle != null ) {
+      XulLabel removeDatasourceConfirmationDialogLabel = (XulLabel) removeDatasourceConfirmationDialog.getElementById( "removeDatasourceConfirmationDialogLabel" );
+      LogicalModelSummary logicalModelSummary = getDialogResult();
```

Comment:
I added a check for the method return instead, since that this way my method is independent from the getDialogResult method implementation.